### PR TITLE
feat(cloudfront): additional cloudfront distribution metrics

### LINKED ISSUE #22922: (cloudfront): Flag on Distribution L2 to enable publishing additional CloudWatch metrics
### Describe the feature

The CloudFront UI and API lets you [enable more granular metrics per-distribution](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/monitoring-using-cloudwatch.html), at an additional (fixed) cost.

There is an L1 construct that lets you do this, which can be easily integrated into the L2 construct and enabled via a prop.

### Use Case

I want to create a dashboard of data like origin latency, Lambda@Edge error rate etc. and alarm on these metrics.

### Proposed Solution

An additional prop for `cloudfront.Distribution`:
```
new Distribution(this, 'Distro', {
    ...,
    publishAdditionalMetrics: true,  // default: false
});
```

If this is set to `true` a `CfnMonitoringSubscription` instance is created:
```
new CfnMonitoringSubscription(nextJsDeployment.distribution, 'MonitoringSubscription', {
    distributionId: nextJsDeployment.distribution.distributionId,
    monitoringSubscription: {
        realtimeMetricsSubscriptionConfig: {
            realtimeMetricsSubscriptionStatus: 'Enabled',
        }
    }
});
```

Also, we probably want to expose these metrics via `distro.metricsXyz()` methods (should probably throw, if `publishAdditionalMetrics` is set to false.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.50.0

### Environment details (OS name and version, etc.)

n/a

### PR DESCRIPTION
**Pull request by badmintoncryer**
This PR supports the activation of [additional cloudfront distribution metrics](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/viewing-cloudfront-metrics.html#monitoring-console.distributions-additional).

```ts
new cloudfront.Distribution(this, 'myDist', {
  defaultBehavior: { origin: new origins.HttpOrigin('www.example.com') },
  publishAdditionalMetrics: true, // added
});
```

Closes #22922.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by GavinZZ** [comment]
One question I have is that this change enabled these additional metrics, but how do users easily access these new metrics? It would be nice to have something similar like L2 Lambda Function to expose the metrics for ease of access. 
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-lambda/lib/function.ts#L688

File: packages/aws-cdk-lib/aws-cloudfront/README.md [hunk 1]
```diff
@@ -584,6 +584,21 @@ new cloudfront.Distribution(this, 'myDist', {
...
+- Origin latency: See the total time spent from when CloudFront receives a request to when it provides a response to the network (not the viewer), for responses that are served from the origin, not the CloudFront cache.
+- Cache hit rate: View cache hits as a percentage of total cacheable requests, excluding errors.
+
+```ts
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: new origins.HttpOrigin('www.example.com') },
+  publishAdditionalMetrics: true,
+});
+```
+
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-cloudfront/README.md:601]: nit: Please add an example on calling the metrics function you added to the README as well.

**Comment by badmintoncryer** [comment]
@GavinZZ Thank you for your review, and I apologize for pushing the content while it was still being revised. 
Based on your feedback, I have made some modifications. Could you please check it again?

**Comment by GavinZZ** [comment]
Changes look good. Happy to approve once the CI passes.

**Comment by badmintoncryer** [comment]
@GavinZZ I've finally passed CI... Sorry for the wait.

**Review by GavinZZ — APPROVED** [review]
Thank you for contributing!

**Pull request merged**